### ISSUE TITLE
Feature request: Show countdown until next folder scan

### ISSUE THREAD

**Issue by eliempje** [syncthing/syncthing#3586]
Currently for a sync folder one can only see the time at witch interval the folder is scanned. E.g. 600 seconds.
Idea is to add a countdown timer.

**Comment by scienmind** [comment]
As in "live display" for the seconds left until the next scan?

While I can see that is can be "kinda nice" to see the internal clocks of it and see the time left, I can't see any practical purpose.
It'll just add extra frequent-changing element to the GUI, taxing the CPU.

If you need to scan now, and wonder if wait until the next scan, just hit "rescan".
Besides, the best practice is to use inotify anyways so that the scanner will be auto-invoked upon file modifications. In that case, I also can't see why would it be useful.

**Comment by canton7** [comment]
JS is pretty good at not doing anything if you're not looking at the tab, and we don't need to hit syncthing, so I can't see the issue.

**Comment by eliempje** [comment]
Ok I get the point

**Comment by calmh** [comment]
Syncthing could just send an "next rescan at: $timestamp" attribute in the state change event, and the gui could do the rest. So this should be fairly simple and clean to do.

**Comment by uok** [comment]
I think a simple (not ticking) info "next scan: (timestamp = last scan time + scan interval)" is helpful

**Comment by calmh** [comment]
We randomly fuzz the scan interval a bit (about +/- 30% iirc), so it needs to be announced after each scan to be reliable. (This is just to avoid folders staying in lockstep with each other)

**Comment by uok** [comment]
correct, I just remembered this too (see https://github.com/syncthing/syncthing/commit/54c3caad530fb16a3764ee70955280f042f2860f)

**Comment by canton7** [comment]
There's the fun around syncthing-inotify randomly delaying scans as well?

**Comment by calmh** [comment]
Yes, but Syncthing knows that and could in fact reflect it in this way. That would make the ui less of a lie on this point than it is now.

**Comment by canton7** [comment]
Ah, no, I guess the GUI gets events when the next scan is delayed, because a scan just happened... duh. Carry on.

**Comment by Ferroin** [comment]
I like the idea of having a time-stamp showing when the next scan will be as opposed to a counter showing remaining time.  Doing the time-stamp would allow a mostly static element instead of one that needs to update every second, and makes more sense when you start talking about scan delays longer than a few minutes.  Having the ability to choose between the two, and possibly also just disable it would be even better (especially considering that people are not likely to agree on which is better).

**Comment by canton7** [comment]
OTOH, you care about how long you have to wait until the next scan. If you're given a timestamp, you're going to be doing that "wait time" calculation yourself in your head - you're not going to say to yourself "OK, the next scan is at 6, I'll come back then".

**Comment by Ferroin** [comment]
That really depends on both the use case and what the rescan interval is.  I've got a couple shared folders which have 15+minute rescan intervals (large data sets, and I'm using syncthing to get a copy for a backup every 15 minutes).  In such a case, I don't care about how long is left, I care when the next scan will happen.  I'm not saying that the countdown is a bad idea, but for my usage at least, the timestamp makes more sense.
